drivers/sensor/sun8i-thermal.c

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright © 2017-2018 The Crust Firmware Authors.
+ * SPDX-License-Identifier: (BSD-3-Clause OR GPL-2.0)
+ */
+
+#include <debug.h>
+#include <error.h>
+#include <limits.h>
+#include <mmio.h>
+#include <sensor.h>
+#include <sensor/sun8i-thermal.h>
+
+#define SENSOR_COUNT                3
+
+#define THS_0_DATA_REGISTER         0x80
+#define THS_1_DATA_REGISTER         0x84
+#define THS_2_DATA_REGISTER         0x88
+
+#define THS_CTL_REGISTER0           0x00
+#define THS_CTL_REGISTER1           0x04
+#define THS_CTL_REGISTER2           0x40
+
+#define THS_FILTER_CONTROL_REGISTER 0x70
+
+enum {
+	THS0,
+	THS1,
+	THS2,
+};
+
+static struct sensor_info sun8i_thermal_sensor_info[SENSOR_COUNT] = {
+	[THS0] = {
+		.name    = "ths0",
+		.reading = UINT32_MAX, /* Symbolizes no reading yet. */
+		.formula = {
+			.min_value  = 680,
+			.multiplier = -8.56,
```

Comment:
Integers!!!! We do not and cannot have floating point anywhere.

Reply:
done

Reply:
Note this doesn't mean truncate and lose precision. It means multiply by a factor large enough that everything is whole numbers (say, 1000).